### PR TITLE
Update transaction details domain UI changes - Closes #4422

### LINKED ISSUE #4422: Update transaction details domain UI changes
### Description

- Implement new [design](https://www.figma.com/file/9JbvSmtLNJIttXp68uhmQR/Change-API-to-support-Lisk-SDK-v6-API?node-id=98%3A15709) changes for transaction details
- Update transaction list view
- Update transaction details view to include events

### Acceptance Criteria

- Users should be able to view transactions
- Users should be able to view transaction details and its events (ordered by eventID)
- Add required unit tests. 

### Additional Information

- Implement event exploration
- Display list of events in order for given transaction
- Event data should be displayed in encoded format initially, and allow users to decode on need basis (based on event schema received from metadata endpoint)

### PR DESCRIPTION
**Pull request by eniolam1000752**
### What was the problem?

This PR resolves #4422

### How was it solved?

- [x] Implement Transaction details layout
- [x] integrate Transaction Events list 
- [x] implement new Transaction details section
- [x] Update unit tests

### How was it tested?

Visually
jest

### CODE REVIEW

File: src/modules/transaction/components/TransactionDetailRow/TransactionDetailRow.test.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      <TransactionDetailRow {...props} data={{ ...props.data, type: 'status', value: 'fail' }} />
+    );
+    expect(screen.queryByText('fail').className).toMatch(/fail/g);
+  });
+
+  it('should render expand button', async () => {
+    render(<TransactionDetailRow {...props} data={{ ...props.data, type: 'expand' }} />);
+
+    expect(screen.getByText('Expand')).toBeTruthy();
+    fireEvent.click(screen.getByText('Expand')).toHaveBeenCalledTimes(1);
```
Comment (ikem-legend) [inline src/modules/transaction/components/TransactionDetailRow/TransactionDetailRow.test.js]: @eniolam1000752 Any reason for asserting like this?

File: src/modules/transaction/components/TransactionDetailRow/components.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+import { useTranslation } from 'react-i18next';
+import Tooltip from 'src/theme/Tooltip';
+import { useTheme } from 'src/theme/Theme';
+import { TertiaryButton } from 'src/theme/buttons';
+import WalletVisual from 'src/modules/wallet/components/walletVisual';
+import { truncateAddress } from 'src/modules/wallet/utils/account';
+import styles from './TransactionDetailRow.css';
+
+export const Text = ({ value, isCapitalized }) => (
+  <div className={`${styles.text} ${isCapitalized ? styles.capitlized : ''} ${grid['col-xs-7']}`}>
```
Comment (ikem-legend) [inline src/modules/transaction/components/TransactionDetailRow/components.js]: @eniolam1000752 Please fix typo in `capitlized`

File: src/modules/transaction/components/TransactionDetailsView/index.js [hunk 3]
```diff
@@ -1,53 +1,152 @@
-import React from 'react';
+import React, { useMemo, useState } from 'react';
 import { isEmpty } from 'src/utils/helpers';
+import ReactJson from 'react-json-view';
+import { useTranslation } from 'react-i18next';
+import { parseSearchParams } from 'src/utils/searchParams';
+import { withRouter } from 'react-router';
```
Comment (ikem-legend) [inline src/modules/transaction/components/TransactionDetailsView/index.js]: @eniolam1000752 It might be best to swap these two lines so that third party imports can be separated from local imports

File: src/modules/transaction/components/TransactionDetailsView/index.js [hunk 4]
```diff
@@ -1,53 +1,152 @@
...
-  transaction: { error, isLoading, data }, wallet,
-  containerStyle,
-}) => {
-  if (!error && isEmpty(data)) {
-    return <div />;
-  }
+const TransactionDetails = ({ location }) => {
+  const transactionId = parseSearchParams(location.search).transactionId;
+  const { t } = useTranslation();
+  const [isParamsCollasped, setIsparamsCollapsed] = useState(false);
```
Comment (ikem-legend) [inline src/modules/transaction/components/TransactionDetailsView/index.js]: @eniolam1000752 Please use camel case here appropriately

File: src/modules/transaction/components/TransactionRow/index.js [hunk 5]
```diff
@@ -1,5 +1,6 @@
 import React from 'react';
 import { withTranslation } from 'react-i18next';
```
Comment (ikem-legend) [inline src/modules/transaction/components/TransactionRow/index.js:2]: @eniolam1000752 `withTranslation` can possibly be replaced with `useTranslation`

File: src/modules/transaction/manager/transactionDetailViewManager.js [hunk 6]
```diff
@@ -4,18 +4,16 @@ import { compose } from 'redux';
 import { connect } from 'react-redux';
 import { withRouter } from 'react-router-dom';
 import { withTranslation } from 'react-i18next';
-import grid from 'flexboxgrid/dist/flexboxgrid.css';
+// import grid from 'flexboxgrid/dist/flexboxgrid.css';
```
Comment (ikem-legend) [inline src/modules/transaction/manager/transactionDetailViewManager.js]: @eniolam1000752 Comments like this should either be removed or have a `TODO` added

File: src/routes/routes.js [hunk 7]
```diff
@@ -160,10 +167,10 @@ export const modals = {
...
     forbiddenTokens: [],
   },
-  transactionDetails: {
-    isPrivate: false,
-    forbiddenTokens: [],
-  },
+  // transactionDetails: {
+  //   isPrivate: false,
+  //   forbiddenTokens: [],
+  // },
```
Comment (ikem-legend) [inline src/routes/routes.js]: @eniolam1000752 I believe this should be removed as it's no longer in use

File: src/modules/transaction/components/TransactionRow/index.js (hunk 5 above)
Comment (eniolam1000752) [reply src/modules/transaction/components/TransactionRow/index.js:2]: I don't want to touch the transaction row since its out of scope for this PR. I think this should be addressed in this issue https://github.com/LiskHQ/lisk-desktop/issues/4423

File: src/modules/transaction/manager/transactionDetailViewManager.js [hunk 8]
```diff
@@ -4,18 +4,14 @@ import { compose } from 'redux';
...
 import { withTranslation } from 'react-i18next';
-import grid from 'flexboxgrid/dist/flexboxgrid.css';
 import { selectActiveTokenAccount, selectActiveToken } from 'src/redux/selectors';
 import { getTransaction } from '@transaction/api';
-import Dialog from '@theme/dialog/dialog';
 import withData from 'src/utils/withData';
 import { parseSearchParams } from 'src/utils/searchParams';
 import TransactionDetails from '../components/TransactionDetailsView';
 
 const WrappedInDialog = (props) => (
```
Comment (ikem-legend) [inline src/modules/transaction/manager/transactionDetailViewManager.js]: @eniolam1000752 Since the Dialog has been removed, this should be renamed appropriately

File: src/utils/testHelpers.js [hunk 9]
```diff
@@ -146,6 +145,28 @@ export const renderWithQueryClient = (Component, props) => {
   return render(
     <QueryClientProvider client={queryClient}>
       <Component {...props} />
-    </QueryClientProvider>,
+    </QueryClientProvider>
+  );
+};
+
+/**
+ * Renders components that are wrapped in QueryClientProvider
```
Comment (ikem-legend) [inline src/utils/testHelpers.js]: @eniolam1000752 Please update documentation to include router

File: src/modules/transaction/components/TransactionDetailRow/TransactionDetailRow.js [hunk 10]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import { AddressWithName, ExpandToggle, Text, Title, TransactionStatus } from './components';
+import styles from './TransactionDetailRow.css';
+
+const TransactionEventRow = ({ data, isParamsCollasped, onToggleJsonView }) => {
```
Comment (ikem-legend) [inline src/modules/transaction/components/TransactionDetailRow/TransactionDetailRow.js]: @eniolam1000752 Please fix typo in `isParamsCollasped` and related usage

File: setup/jest.config.js [hunk 11]
```diff
@@ -304,6 +304,7 @@ module.exports = {
     'src/modules/account/hooks/useCurrentAccount.js',
     'src/modules/transaction/store/actions.js',
     'src/modules/transaction/hooks/useCreateTransaction.js',
+    'src/modules/transaction/components/TransactionDetails', // @Todo: would resolve this when all transation implentations are finallized
```
Comment (ikem-legend) [inline setup/jest.config.js]: @eniolam1000752 Please fix typos here

**Review by ikem-legend — COMMENTED** [review]
LGTM. Just some minor changes left

File: package.json [hunk 12]
```diff
@@ -116,7 +116,7 @@
     "@hookform/resolvers": "2.8.8",
     "@ledgerhq/hw-transport": "6.3.0",
     "@ledgerhq/hw-transport-node-hid": "6.6.0",
-    "@liskhq/lisk-client": "6.0.0-alpha.32217",
+    "@liskhq/lisk-client": "6.0.0-alpha.32056",
```
Comment (soroushm) [inline package.json]: revert the version change in order to pass unit test

File: src/modules/common/components/amountField/Heading/Heading.js [hunk 13]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { TertiaryButton } from 'src/theme/buttons';
+import Icon from 'src/theme/Icon';
+import styles from './Heading.css';
+
+const Heading = ({ title, onGoBack, history, noBackButton, className }) => (
+  <div className={`${styles.wrapper} ${className}`}>
+    {!noBackButton && (
```
Comment (soroushm) [inline src/modules/common/components/amountField/Heading/Heading.js:8]:
```suggestion
    {typeOf onGoBack  === 'function' && (
```

File: src/modules/transaction/components/TransactionDetailsView/index.js [hunk 14]
```diff
@@ -1,53 +1,154 @@
...
+
+  const {
+    data: transactions,
+    error,
+    isLoading,
+    isFetching,
+  } = useTransactions({
+    config: {
+      params: {
+        id: transactionId,
```
Comment (soroushm) [inline src/modules/transaction/components/TransactionDetailsView/index.js]: {"error":true,"message":"Unknown input parameter(s): id"}
```suggestion
        transactionID: transactionId,
```

**Review by soroushm — CHANGES_REQUESTED** [review]

File: src/modules/transaction/components/TransactionRow/index.js [hunk 15]
```diff
@@ -24,7 +25,8 @@ const TransactionRow = ({
     <DialogLink
```
Comment (soroushm) [inline src/modules/transaction/components/TransactionRow/index.js]: as it's just a link and data is a params shall we use Link insted of dialogLink ?

File: src/modules/common/components/amountField/Heading/Heading.js (hunk 13 above)
Comment (eniolam1000752) [reply src/modules/common/components/amountField/Heading/Heading.js:8]: this would be tricky because if `onGoBack` isn't a function what it means is that i would always have to provide onGoBack as a function for every simple goback functionality which can easily be handled in the component been wrapped with `withRouter`

File: src/modules/transaction/components/TransactionRow/index.js (hunk 15 above)
Comment (eniolam1000752) [reply src/modules/transaction/components/TransactionRow/index.js]: I felt I should not change the original component used (but just extend it) since it would be outside the scope of this PR

File: package.json [hunk 16]
```diff
@@ -296,4 +296,4 @@
       "pre-commit": "pretty-quick --staged"
     }
   }
-}
+}
```
Comment (ikem-legend) [inline package.json]: @eniolam1000752 New line is missing here

File: src/modules/transaction/components/TransactionDetailRow/TransactionDetailRow.test.js [hunk 17]
```diff
@@ -0,0 +1,82 @@
...
+}));
+
+describe('TransactionDetailRow', () => {
+  const props = {
+    data: {
+      tooltip: 'test-tooltip',
+      value: 'test value',
+      label: 'test-title',
+    },
+    isParamsCollasped: false,
```
Comment (ikem-legend) [inline src/modules/transaction/components/TransactionDetailRow/TransactionDetailRow.test.js]: @eniolam1000752 Please fix typo here because there's a mismatch between the props for the test and the props for the component

**Review by ikem-legend — DISMISSED** [review]

File: src/modules/transaction/components/TransactionDetailRow/TransactionDetailRow.css [hunk 18]
```diff
@@ -0,0 +1,167 @@
...
+.container {
+  display: flex;
+  width: 100%;
+  position: relative;
+
+  & > div:first-child {
+    padding: 20px 22px 20px 0px;
+    color: var(--color-content-light);
+    font-weight: var(--font-weight-normal);
+    font-size: 14px;
```
Comment (reyraa) [inline src/modules/transaction/components/TransactionDetailRow/TransactionDetailRow.css]: We have a set of mixins that help us ensure we have a limited predefined set of typographical elements. I'd recommend using them.

File: src/modules/transaction/components/TransactionDetailRow/TransactionDetailRow.css [hunk 19]
```diff
@@ -0,0 +1,167 @@
...
+    text-overflow: ellipsis;
+    font-weight: var(--font-weight-semi-bold);
+    font-size: var(--font-size-h6);
+  }
+}
+
+.expandToggle {
+  --content-font: 'basier-circle';
+
+  padding: 0px 5px !important;
```
Comment (reyraa) [inline src/modules/transaction/components/TransactionDetailRow/TransactionDetailRow.css]: If you use nesting correctly, not only you don't need to prioritize using `!important` but also you prevent side effects.

File: src/modules/transaction/components/TransactionDetailRow/TransactionDetailRow.css [hunk 20]
```diff
@@ -0,0 +1,167 @@
...
+      background-color: var(--color-active-tab);
+    }
+
+    &.dark {
+      color: var(--color-ultramarine-blue);
+      background-color: var(--color-cloud-burst);
+    }
+  }
+
+  &.fail {
```
Comment (reyraa) [inline src/modules/transaction/components/TransactionDetailRow/TransactionDetailRow.css:149]: You can use variables and re-define them using 
```
:root [data-theme="dark"] {}
```

So you don't have to add `.dark` and `.light` class names

File: src/modules/transaction/components/TransactionDetailsView/index.js [hunk 21]
```diff
@@ -1,53 +1,154 @@
...
+        isCapitalized: true,
+      },
+      {
+        label: t('Sender'),
+        value: sender,
+        type: 'address',
+      },
+      {
+        label: t('Transaction Fee'),
+        value: <TokenAmount val={fee} token="LSK" />,
```
Comment (reyraa) [inline src/modules/transaction/components/TransactionDetailsView/index.js]: Can we make the token name dynamic?

File: src/modules/transaction/components/TransactionDetailsView/index.js [hunk 22]
```diff
@@ -1,53 +1,154 @@
...
+      id,
+      moduleCommandName,
+      sender = {},
+      nonce,
+      fee,
+      block = {},
+      confirmations,
+      executionStatus,
+    } = transaction;
+    const [txModule, txType] = moduleCommandName.split(':');
```
Comment (reyraa) [inline src/modules/transaction/components/TransactionDetailsView/index.js]: There's a utility for this. I know it might seem like over engineering, but we've gone through phases where drastic changes were applied to the API responses. Using utilities help us reduce the number of places we have to touch in the future and prevent regressions.

**Review by reyraa — DISMISSED** [review]

File: src/modules/transaction/components/TransactionDetailsView/index.js (hunk 21 above)
Comment (eniolam1000752) [reply src/modules/transaction/components/TransactionDetailsView/index.js]: so I have added to todo to this. since we don't have this parameter as part of the response payload for now @soroushm suggested we leave it like this and add a todo to it

File: src/modules/transaction/components/TransactionDetailRow/TransactionDetailRow.css (hunk 20 above)
Comment (eniolam1000752) [reply src/modules/transaction/components/TransactionDetailRow/TransactionDetailRow.css:149]: done

File: src/modules/transaction/components/TransactionDetailRow/TransactionDetailRow.css (hunk 18 above)
Comment (eniolam1000752) [reply src/modules/transaction/components/TransactionDetailRow/TransactionDetailRow.css]: done

File: src/modules/transaction/components/TransactionDetailRow/TransactionDetailRow.css (hunk 19 above)
Comment (eniolam1000752) [reply src/modules/transaction/components/TransactionDetailRow/TransactionDetailRow.css]: Thanks for the suggestion. I think `!important` is needed here to override this property

**Review by soroushm — DISMISSED** [review]

**Review by ikem-legend — DISMISSED** [review]

File: src/theme/dialog/link.js [hunk 23]
```diff
@@ -1,13 +1,18 @@
...
   const linkEl = useRef(null);
   const onClick = () => {
-    addSearchParamsToUrl(history, { modal: component, ...data });
+    if(!path) {
+      addSearchParamsToUrl(history, { modal: component, ...data });
+    } else {
+      const newSearchString = appendSearchParams(history.location.search, data);
+      history.push(`${path}${newSearchString}`);
+    }
   };
```
Comment (soroushm) [inline src/theme/dialog/link.js]: @eniolam1000752 as its not used anymore could you please revert the changes

**Review by soroushm — APPROVED** [review]

**Review by ikem-legend — APPROVED** [review]
